models/movie.js

```diff
@@ -136,12 +98,9 @@ function transformJsonMovie(doc, json, options) {
   delete json._id;
   delete json.__v;
 
-  if (json.director instanceof ObjectId) {
-    // Remove the director property by default (there's a "directorHref" virtual property)
-    delete json.director;
-  } else {
+  if (!(json.directorId instanceof ObjectId)) {
     // If the director was populated, include it in the serialization
-    json.director = doc.director.toJSON();
+    json.directorId = doc.directorId.toJSON();
```

Comment:
```suggestion
    json.director = doc.directorId.toJSON();
```

The idea is that the `directorId` is always in the response, but the client can ask to also fetch and retrieve the whole `director` object along with it. It should not replace the `directorId`.